src/Clapi/Valuespace.hs

```diff
@@ -395,6 +395,7 @@ processToRelayClientDigest reords dd vs =
 data ValidationErr
   = GenericErr String
   | ProgrammingErr String
+  | BadNodeType String
```

Comment:
We _could_ make an enumeration of the rose tree node types and then put one of those values in this error message. I'm not sure the string is actually that useful!